app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala

```diff
@@ -196,4 +196,61 @@ class RecordsRepository @Inject() (
       case None        => Future.successful(Seq.empty)
     }
   }
+
+  def searchTermFilter(searchTerm: Option[String]): Bson =
+    searchTerm match {
+      case Some(value) =>
+        val escapedValue = escapeRegexSpecialChars(value)
+        Filters.or(
```

Comment:
Filter endpoint is currently used to check for unique traderRef values https://github.com/hmrc/trader-goods-profiles-frontend/blob/main/app/controllers/goodsRecord/TraderReferenceController.scala#L89
Since we need both partial search and exact match, can’t we use the same method? Check out this implementation for reference https://github.com/hmrc/trader-goods-profiles-data-store/pull/89/files#diff-d92031c945d27f64e7fe6365e45440459788e83da500fba93390bcc6e8d0f280R86

Reply:
I see what you mean, ideally I think we should have a new separate endpoint for checking if the trader ref is unique so it's not coupled in with the filtering stuff and then we can look to clean up the old filtering code and will be able to delete the exact match search term as that's an extra parameter we don't necessarily need anymore

Reply:
Agreed. Let's handle it separately then